### ISSUE TITLE
Can't authenticate with AWS EKS SSO

### ISSUE THREAD

**Issue by curbol** [vscode-kubernetes-tools/vscode-kubernetes-tools#1168]
Sounds like a super useful extension, but I'm not able to authenticate to my cluster in the extension. My company's cluster is in AWS EKS and we use SSO login. I can authenticate from a terminal and access the cluster with kubectl in the CLI, but the extension says 'Unable to locate credentials. You can configure credentials by running "aws configure".' I'm thinking it doesn't work with SSO?

After I choose the `/.kube/config` file I get this error message in VS code:

> Unable to locate credentials. You can configure credentials by running "aws configure". Unable to locate credentials. You can configure credentials by running "aws configure". Unable to locate credentials. You can configure credentials by running "aws configure". Unable to locate credentials. You can configure credentials by running "aws configure". Unable to locate credentials. You can configure credentials by running "aws configure". Unable to connect to the server: getting credentials: exec: executable aws failed with exit code 253

<img width="439" alt="image" src="https://user-images.githubusercontent.com/5971687/231647589-414d6410-9351-4522-91ff-0d1a86d50f14.png">

Here's showing things working in my CLI
```bash
~
❯ awsdev
Added new context arn:aws:eks:us-east-2:0000000000:cluster/development-shared-infra-cluster to /Users/curtis/.kube/config

~
❯ kubectl get pods -n noms 
NAME                                 READY   STATUS    RESTARTS   AGE
alb-noms-5bc69684db-k4h4n            2/2     Running   0          97m
alb-noms-5bc69684db-q4fmw            2/2     Running   0          97m
alb-noms-5bc69684db-qn7jp            2/2     Running   0          97m
```

If there was a way to have the extension run my `awsdev` script to authenticate I expect everything would work fine. Maybe a feature could be added to set extension start-up scripts?

**Comment by curbol** [comment]
> I don't have a real experience with this but it looks like `awsdev` adds the context to the kubeconfig file and kubectl should have all infos just by looking at it, no? The things that i can think of are, are you sure the extension is targetting the same kubeconfig? Are you using the same kubectl (maybe a different version could create some misbehavior)?
> 
> I'll try to get my hands on a similar scenario to try it. Sorry for not having a solution atm.

Thanks for the response! I hadn't thought of the possibility of there being two versions of kubectl. I'll investigate that.

Our DevOps team created the `awsdev` command. This is what it's doing if that helps:
```bash
# in ~/.zshrc
alias awsdev='export AWS_REGION=us-east-2 && aws-vault exec platform-development -- bash -c "rm -rf ~/.kube/* && aws eks update-kubeconfig --name development-shared-infra-cluster --region us-east-2 && zsh"'
```

```bash
# in ~/.aws/config
[profile platform-development]
sso_start_url = https://omitted.awsapps.com/start
sso_region = us-east-1
sso_account_id = 000000000000
sso_role_name = AdministratorAccess
region = us-east-2
```

**Comment by curbol** [comment]
I was able to fix my issue by launching vs-code from the same terminal that I authenticated with AWS in.

For example:
```
~
❯ awsdev

~
❯ code
```

1. `awsdev` gives permissions to `kubectl`, but only for the current terminal window.
2. `code` launches vs-code from the current terminal window which allows the vs-code kubernetes extension to use the authorized `kubectl`

It's not ideal, but works fine for me.
